internal/statemachine/classic_states.go

```diff
@@ -300,6 +304,23 @@ func (stateMachine *StateMachine) calculateStates() error {
 		}
 	}
 
+	// only run makeDisk if there is an artifact to make
+	if classicStateMachine.ImageDef.Artifacts.Qcow2 != nil {
+		// only run make_disk once
+		found := false
+		for _, stateFunc := range rootfsCreationStates {
+			if stateFunc.name == "make_disk" {
+				found = true
+			}
+		}
+		if !found {
+			rootfsCreationStates = append(rootfsCreationStates,
+				stateFunc{"make_disk", (*StateMachine).makeDisk})
+		}
+		rootfsCreationStates = append(rootfsCreationStates,
+			stateFunc{"create_qcow2_image", (*StateMachine).createQcow2})
```

Comment:
Could I recommend using `make_qcow2_disk` or `make_disk_qcow2` or `make_qcow2_image` as the state names for the qcow2 states? Since `make_disk` is the state responsible for .img creation, maybe it would be better to follow a similar naming convention?

Reply:
Sure, no problem!